integration/tests/runtime-workflow-datasync.ts

```diff
@@ -0,0 +1,245 @@
+        lastSync
+    }
+}
+`
+
+beforeAll(async () => {
+  try {
+    mkdirSync("./output-datasync-mongo");
+    mkdirSync("./output-datasync-mongo/client")
+
+    mongoClient = new MongoClient('mongodb://mongodb:mongo@localhost:27017/users?authSource=admin', { useUnifiedTopology: true });
+    await mongoClient.connect();
+    db = mongoClient.db('users');
+    graphbackApi = buildGraphbackAPI(modelText, {
+      dataProviderCreator: createDataSyncMongoDbProvider(db),
+      serviceCreator: createDataSyncCRUDService({ pubSub: new PubSub() }),
+      plugins: [
+        new SchemaCRUDPlugin({ outputPath: "./output-datasync-mongo/schema/schema.graphql" }),
+        new ClientCRUDPlugin({ outputFile: './output-datasync-mongo/client/graphback.graphql' }),
+        new DataSyncPlugin()
+      ]
+    });
+    
+    await seedDatabase(db);
+    
+    const source = await loadDocuments(path.resolve(`./output-datasync-mongo/client/graphback.graphql`), {
+      loaders: [
+        new GraphQLFileLoader()
+      ]
+    });
+    documents = source[0].document;
+  } catch (e) {
+    throw e;
+  }
+})
+
+beforeEach(() => {
+  const { typeDefs, resolvers, contextCreator } = graphbackApi;
+  server = new ApolloServer({
+    typeDefs,
+    resolvers,
+    context: contextCreator
+  });
+  
+  client = createTestClient(server);
+})
+
+afterEach(() => server.stop())
+
+afterAll(async () => {
+  rmdirSync(path.resolve('./output-datasync-mongo'), { recursive: true });
+  await db.dropCollection('note');
+
+  return mongoClient.close();
+});
+
+async function seedDatabase(db: Db) {
+  const notes = [
+    {
+      title: 'Note A',
+      description: 'Note A Description',
+      [DataSyncFieldNames.lastUpdatedAt]: startTS.valueOf(),
+      [DataSyncFieldNames.deleted]: false
+    },
+    {
+      title: 'Note B',
+      description: 'Note B Description',
+      [DataSyncFieldNames.lastUpdatedAt]: startTS.valueOf(),
+      [DataSyncFieldNames.deleted]: false
+    }
+  ]
+
+  for (const note of notes) {
+    const { ops } = await db.collection("note").insertOne(note);
+    notesId.push(ops[0]._id);
+  }
+
+  
+}
+
+test('Sync all notes', async () => {
+
+  const { data, errors, extensions } = await client.query({ operationName:"syncNotes", query: SYNC_NOTES , variables: {
+    lastSync: 0,
+    filter: {}
+  }});
+  expect(data).toBeDefined();
+  expect(data.syncNotes?.items).toEqual([
+    {
+      _id: notesId[0],
+      title: 'Note A',
+      description: 'Note A Description',
+      [DataSyncFieldNames.lastUpdatedAt]: startTS.valueOf(),
+      [DataSyncFieldNames.deleted]: false
+    },
+    {
+      _id: notesId[1],
+      title: 'Note B',
+      description: 'Note B Description',
+      [DataSyncFieldNames.lastUpdatedAt]: startTS.valueOf(),
+      [DataSyncFieldNames.deleted]: false
+    }
+  ])
+})
+
+test('Sync all notes with filtering', async () => {
+
+  const { data } = await client.query({ query: SYNC_NOTES , variables: {
+    lastSync: 0,
+    filter: {
```

Comment:
Are we planning to add `limit`

Reply:
@wtrocki There was some discussion about this: https://github.com/aerogear/graphback/issues/1724